### ISSUE TITLE
[Feature]: Annotation editors are not exposed

### ISSUE THREAD

**Issue by Stamo-Gochev** [mozilla/pdf.js#18656]
### Is the feature relevant to the Firefox PDF Viewer?

No

### Feature description

I am considering using PDF.js for creating a custom PDF Viewer by using as much bits from PDF.js as possible.

In particular, I want to import the annotations functionality and customize it.

However, the files in https://github.com/mozilla/pdf.js/tree/master/src/display/editor are not exported in https://github.com/mozilla/pdf.js/blob/master/src/pdf.js#L66-L68

Is this expected? 

Can the files be also re-exported, so that they can be imported and further customized?

**Comment by Snuffleupagus** [comment]
> Is this expected? If yes - why?

Yes, since the official API (mostly) only exposes the functionality needed in the default viewer.

> Can the files be also re-exported, so that they can be imported and further customized?

Sorry, but we can't/won't just expose all of that functionality since that'd increase maintenance/support burden way too much and would also make it much more difficult for us to make future changes to that code.
(Note that the editor-code is under very active development, and there's also e.g. issue #18625 which seems somewhat related.)
